sdk/storage/inc/datalake/datalake_responses.hpp

```diff
@@ -10,26 +10,27 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
 
   // ServiceClient models:
 
-  using UserDelegationKey = Blobs::GetUserDelegationKeyResult;
-  using ListFileSystemsResult = ServiceListFileSystemsResponse;
+  using GetUserDelegationKeyResult = Blobs::GetUserDelegationKeyResult;
+  using ListFileSystemsSegmentResult = ServiceListFileSystemsResult;
 
   // FileSystemClient models:
 
-  using FileSystemDeleteInfo = FileSystemDeleteResponse;
-  using ListPathsResult = FileSystemListPathsResponse;
+  using DeleteFileSystemResult = FileSystemDeleteResult;
+  using ListPathsResult = FileSystemListPathsResult;
 
-  struct FileSystemProperties
+  struct GetFileSystemPropertiesResult
   {
     std::string ETag;
     std::string LastModified;
     std::map<std::string, std::string> Metadata;
   };
 
-  using FileSystemInfo = FileSystemCreateResponse;
+  using CreateFileSystemResult = FileSystemCreateResult;
+  using SetFileSystemMetadataResult = FileSystemCreateResult;
 
   // PathClient models:
 
-  using PathDeleteInfo = PathDeleteResponse;
+  using DeletePathResult = PathDeleteResult;
```

Comment:
what? cannot you just name in `DeletePathResult` in protocol layer?

Reply:
I cannot.
E.g. for `ShareCreateSnapshotResult`, it comes from 3 parts: `Share` is the service name, `CreateSnapshot` is what swagger defines as the method's name and `Result` is appended by the generator.
If I changed the protocol layer to `DeletePathResult`, then `ShareCreateSnapshotResult` will be `CreateSnapshotShareResult`, which is way worse than the current name.

Reply:
@JeffreyRichter said this naming rule only applies to convenience layer. So I guess we can name protocol layer variables whatever we like.